src/queries/a11yState.ts

```diff
@@ -10,6 +10,23 @@ import type {
   QueryByQuery,
 } from './makeQueries';
 
+export function matchAccessibilityState(
+  node: ReactTestInstance,
+  matcher: AccessibilityState
+) {
+  const stateProp = node.props.accessibilityState;
+
+  // busy, disabled & selected states default to false,
+  // while checked & expended states treat false and default as sepatate values
+  return (
+    matchState(stateProp?.busy ?? false, matcher.busy) &&
+    matchState(stateProp?.disabled ?? false, matcher.disabled) &&
+    matchState(stateProp?.selected ?? false, matcher.selected) &&
+    matchState(stateProp?.checked, matcher.checked) &&
+    matchState(stateProp?.expanded, matcher.expanded)
+  );
+}
+
 function matchState(value: unknown, matcher: unknown) {
```

Comment:
If my understanding is correct , the following will always be true : 

```ts

  expect(view.queryAllByA11yState({ checked: undefined })).toBeTruthy();

```

Is this the intended behaviour ? If checked : undefined results in a state that's different from check: true or checked : false I guess this shouldn't work in any case, however i'm not sure this query is relevant

Reply:
`undefined` is similar to not having given entry when you get the prop from the object: `state.checked`. We could use `hasOwnProp` or something similar but not sure if its worth the effort.

In regular tests you want to check if given checkbox is checked or not checked (or mixed state). I do not see value in querying `undefined` (not set) state.

The goal of this PR is to make `*ByA11yState` queries have the same behavior as we add to `*ByRole` in #1134, where you query by role, name and state at the same time. From that point of view, you would make example queries:

```ts
getByRole('button', { name: "OK", disabled: false }) // <- you want to match implicitly enable buttons
getByRole('checkbox', { name: "Mr", checked: true/false }) // <- you want to examine checkbox state
getByRole('whatever', { name: "Details", expanded: true/false }) // <- you want to examine expandable section state
```

Which would translate to following `*ByA11yState` queries:

```ts
getByA11yState({ disabled: false }) // <- you want to match implicitly enable buttons
getByA11yState({ checked: true/false }) // <- you want to examine checkbox state
getByA11yState({ expanded: true/false }) // <- you want to examine expandable section state
```

IMO in both cases querying for `checked: undefined` or `expanded: undefined` is not useful for the user.

@pierrezimmermannbam do you think this is reasonable from users POV?

Reply:
Yes you're probably right this is probably not a valid use case and not worth supporting. An alternative could be to change matcher typing so that querying {{checked: undefined}} is not allowed. This would make sense if we consider this not relevant and don't support it and it would be less complicated to implement. I'm still not sure it's worth either though, current implementation is also fine to me

Reply:
Afaik using TS optional type: `{ checked?: boolean | 'mixed' }` is allowing both omitting the prop (`{}`) and using `{ checked: undefined }`. How could we allow omitting it, because you want to query other state props, without allowing passing explicit `undefined`?

Reply:
I made a few tries but I couldn't find a way to make this work. Still I'd be surprised that user query for undefined accessibility state so I think we can stick to current behavior